src/main/java/com/rcv/GuiApplication.java

```diff
@@ -29,6 +34,23 @@ public void start(Stage window) throws Exception {
     Parent root = FXMLLoader.load(getClass().getResource("/GuiConfigLayout.fxml"));
     window.setTitle("Universal RCV Tabulator");
     window.setScene(new Scene(root));
+    GuiContext context = GuiContext.getInstance();
+    // cache this so we can parent file chooser to it
+    context.mainWindow = window;
+    EventHandler<WindowEvent> windowEventEventHandler = event -> {
```

Comment:
Is this even necessary now that the window is properly modal?

Reply:
I'm going to try just always doing this aggressive Platform.exit behavior (regardless if the Chooser is open) and see if that causes any problems.  If not, there is no need to cache the Chooser in the GUIContext, and with your other suggestion below there would be no need to change the GUIContext at all yet.

Reply:
this seems to work fine.